WIP Test load ->dump ->load

kidd opened the pull request:
Integrate test-load-and-dump into the regular test suite

camsaul commented on test/metabase/db/load_and_dump_test.clj:
```diff
@@ -0,0 +1,51 @@
+(ns metabase.db.load-and-dump-test
```
Test namespaces should be `ORIGINAL-NAMESPACE-test` e.g. `metabase.cmd.load-from-h2-test`. Tools like CIDER know that namespaces are associated with one another. 

For places that don't really have a corresponding match you should still try to put them in the same folder so I would name this `metabase.cmd.load-and-dump-test`

camsaul commented on test/metabase/db/load_and_dump_test.clj:
```diff
@@ -0,0 +1,51 @@
+
+      ;; XXX how the :mysql works if I don't have it running?
+      (mt/test-drivers #{:postgres :mysql}
+
```
https://guide.clojure.style/#no-blank-lines-within-def-forms

I think there are a few exceptions:
- breaking out a giant `deftest` form into a few different `testing` sections
- breaking out different arities of a `defn` form (this is used as a "bad" example but I don't think it's that bad)

either way, there shouldn't be any blank lines here

kidd committed:
Add test for load from h2 and dump to h2.

Move the test from `bin/test-load-and-dump.sh` inside the regular test
suite.

When testing with h2 driver as the destination db, special case the
jdbc spec.

H2 being in memory makes that the usual jdbc connection that would be
created would be ephemeral, so when doing the dump, and comparing
again against the first, the db would be gone. Adding a CLOSE_DELAY=10
makes it stick for a little while so we can do the comparison.

Not calling `tx/create-db!` in that case because in h2, db is always
created already.

camsaul commented on test/metabase/cmd/load_and_dump_test.clj:
```diff
@@ -0,0 +1,38 @@
+(deftest load-and-dump-test
+  (testing "loading data to h2 and porting to DB and migrating back to H2"
+    (let [h2-fixture-db-file (abs-path "frontend/test/__runner__/test_db_fixture.db")
+          h2-file (abs-path "frontend/test/__runner__/test_db_fixture2.db")
```
for the destination file I'd just put it in `/tmp` so if you run this locally you don't end up accidentally committing it

kidd replied:
ah, great. it was my first approach but then thought that maybe in circle-ci the `/tmp/...` paths would collide if different jobs run in parallel.  But it seems it's fine, so I'll change it.

camsaul reviewed (approved):
Cool, this looks good to me 👍
